### ISSUE TITLE
Enable / disable form in configuration

### ISSUE THREAD

**Issue by steevanb** [symfony/symfony#24269]
| Q                | A
| ---------------- | -----
| Bug report?      | yes
| Feature request? | no
| BC Break report? | yes
| RFC?             | no
| Symfony version  | 3.3.*

Hi !

In app/config/config.yml, this 2 configurations should have same behavior :
```yml
framework:
    form: ~
```
```yml
framework:
    # no form key
```

But first one enable form, and second one, disable it (`framework.form.enabled` default value is `false` : https://symfony.com/doc/current/reference/configuration/framework.html#reference-form-enabled)

Or maybe `~` is not exactly same as "default values" ?
In this case, how can we know differences between `~` and default values  ?

**Comment by chalasr** [comment]
This is a documentation issue, as of 3.3 the `framework.form.enabled` config option is default `true` [if you are requiring `symfony/symfony`](https://github.com/symfony/symfony/blob/master/src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php#L120).

**Comment by chalasr** [comment]
See https://github.com/symfony/symfony-docs/issues/8412

**Comment by steevanb** [comment]
So, i don't understand why `form: ~` and `no form key` does not have same behavior ?

**Comment by chalasr** [comment]
Just tried out, not adding the `form` key at all gives `enabled: false` at end. ping @ogizanagi

**Comment by steevanb** [comment]
When i dump Configuration object, i have 2 properties : `$defaultValue = false`, and `$default = true`.

I'm not sure about it, but maybe when you define form key, default value is readed in `$default`, not `$defaultValue` ?

**Comment by ogizanagi** [comment]
If your using the fullstack (`symfony/symfony`), the `framework.form` node is built using `canBeEnabled` (for BC reasons, as the original way before the framework `Configuration` object was tweaked in #21196 was to have everything disabled by default).

So

```php
framework:
    form: ~
```

is not the same as

```
framework:
    # no form key
```

because `canBeEnabled` [treats `null` as `['enabled' => true]`](https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Config/Definition/Builder/ArrayNodeDefinition.php#L284). The `~` char in yaml does not mean "take default values". It's just `null`. What is really configured behind depends on the way the node was built.

#21196 changed the framework `Configuration` object for Flex projects in order to cleverly auto-enable available features according to the components you've installed. So given you're using flex in your project instead of the fullstack and have the `symfony/form` component installed, `canBeDisabled` is used, which means no `framework.form` key is even required, as [it defaults to `['enabled' => true]` when not set](https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Config/Definition/Builder/ArrayNodeDefinition.php#L311).

So in conclusion, I see no issue here, unless I missed something.

Thanks for opening this though. We should probably upgrade the documentation to explain how defaults behaves with Flex. https://github.com/symfony/symfony-docs/issues/8412 will be a good start.

**Comment by steevanb** [comment]
Ok, to me, that's a bug.

But if you don't think so ...
